src/api/app/models/assignment.rb

```diff
@@ -20,7 +20,7 @@ class Assignment < ApplicationRecord
 
   #### Validations macros
   validate :assignee do
-    errors.add(:assignee, 'must be in confirmed state') unless assignee.state == 'confirmed'
+    errors.add(:assignee, 'must be in confirmed state') unless assignee && assignee.state == 'confirmed'
```

Comment:
What about prepending a validation for testing whether the associated object itself is present? It should be something like: `validates :assignee, presence: true`. Not tested, though.

Reference: https://guides.rubyonrails.org/v7.1/active_record_validations.html#presence

Reply:
It's a bit redundant... Looks like the validation is performed already:
```
[1] pry(main)> a = Assignment.new
=> #<Assignment:0x00007fed1d944348
 id: nil,
 assignee_id: nil,
 assigner_id: nil,
 package_id: nil,
 created_at: nil,
 updated_at: nil>
[2] pry(main)> a.valid?
  Assignment Exists? (0.2ms)  SELECT 1 AS one FROM `assignments` WHERE `assignments`.`package_id` IS NULL LIMIT 1
=> false
[3] pry(main)> a.errors.full_messages
=> ["Assignee must exist", "Assigner must exist", "Package must exist", "Assignee must be in confirmed state"]
```

Reply:
> It's a bit redundant... Looks like the validation is performed already:

True, adding the `validates :assignee, presence: true` is redundant. And it doesn't prevent the `assignee.state == 'confirmed'` validation from being performed. :+1: